src/python/pants/init/extension_loader.py

```diff
@@ -75,58 +79,68 @@ def load_plugins(build_configuration, plugins, working_set):
         if dep.key not in loaded:
           raise PluginLoadOrderError(f'Plugin {plugin} must be loaded after {dep}')
 
-    if 'build_file_aliases' in entries:
-      aliases = entries['build_file_aliases'].load()()
-      build_configuration.register_aliases(aliases)
-
-    if 'register_goals' in entries:
-      entries['register_goals'].load()()
-
-    if 'global_subsystems' in entries:
-      subsystems = entries['global_subsystems'].load()()
-      build_configuration.register_optionables(subsystems)
-
-    if 'rules' in entries:
-      rules = entries['rules'].load()()
-      build_configuration.register_rules(rules)
-
+    if is_v1_plugin:
+      if 'register_goals' in entries:
+        entries['register_goals'].load()()
+
+      # TODO: Might v2 plugins need to register global subsystems? Hopefully not.
```

Comment:
This should be specifically supplanted by `optionable_rule()`, right?

Reply:
Yea, it is.